src/Sylius/Component/Payment/Model/PaymentRequest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Payment\Model;
+
+use Sylius\Component\Resource\Model\TimestampableTrait;
+
+class PaymentRequest implements PaymentRequestInterface
+{
+    use TimestampableTrait;
+
+    protected ?string $hash = null;
+
+    protected ?PaymentMethodInterface $method = null;
+
+    protected ?PaymentInterface $payment = null;
+
+    protected string $state = PaymentRequestInterface::STATE_NEW;
+
+    protected string $type = PaymentRequestInterface::DATA_TYPE_CAPTURE;
+
+    protected mixed $requestPayload = null;
```

Comment:
I think about using an array or a serialized object here, that's why mixed is required here.

- For **non** Payum payment method, you can choose to send what ever you wanted, preferably some serializable data like a json object or array.
- For Payum payment method, a specific object would be more appropriate since the Payum Token requires some after url and other data.